### DISCUSSION TITLE
Problem with access control

### DISCUSSION THREAD

**Discussion by varazir** [authelia/authelia#3805] [Q&A] [answered]
Hello, 

I'm running authelia behind pfSense with HA proxy. 
I have gotten it to work find with standard login bypass/one/two login. 

I wanted to do some more controlled access control, like SSO from my LAN clients. 

The logs file shows this: 
```
time="2022-08-07T07:30:38Z" level=debug msg="Check authorization of subject username= groups= ip=10.0.0.1 and object https://ha.home.dmz/json.htm?type=command&param=getSunRiseSet (method )."
time="2022-08-07T07:30:38Z" level=info msg="Access to https://ha.home.dmz/json.htm?type=command&param=getSunRiseSet (method unknown) is not authorized to user <anonymous>, responding with status code 401" method=HEAD path=/api/verify remote_ip=10.0.0.1
```

The 10.0.0.1 is pfsense interface. My auth server is not on the same subnet/interface as my LAN 
I have the same issue if I set up a bypass when I'm accessing my services locally(from LAN). 
Works if I login but I don't want that locally.  

I see this when I have an OK login so my auth server does get the IP during the login. 
```
time="2022-08-07T07:33:45Z" level=info msg="Access to https://service.home.wtf/ (method unknown) is not authorized to user <anonymous>, responding with status code 401" method=HEAD path=/api/verify remote_ip=10.0.0.1
time="2022-08-07T07:33:59Z" level=debug msg="Mark 1FA authentication attempt made by user 'user1'" method=POST path=/api/firstfactor remote_ip=194.234.110.226
time="2022-08-07T07:33:59Z" level=debug msg="Successful 1FA authentication attempt made by user 'user1'" method=POST path=/api/firstfactor remote_ip=194.234.110.226
```

```
  networks:
    - name: external
      networks:
        - '111.178.116.149'
      
   - domain:
        - "server1.home.dmz"
      policy: bypass
      networks:
        - 'external'

```
Is there a way I can see more  logs ?

**Comment by james-d-elliott** [comment, marked as answer]
Problem is most likely this here or heavily related: https://github.com/authelia/authelia/issues/3346

**Reply by varazir** [reply]
Thanks, worked perfekt

**Reply by josue-soares** [reply]
Hello. I'm using nginx (proxy). In this case what solution sugest?

**Reply by james-d-elliott** [reply]
If the IP in the logs is incorrect then the X-Forwarded-For header is incorrect or absent. In the haproxy case there was a special reason for it being incorrect. In most instances a configuration lacks this header, or the proxy does not have the correct information due to another proxy in the chain or NAT hairpin.

**Reply by josue-soares** [reply]
James, thanks for your answer, in fact I'm behind a NAT but I tested using o authelia server, using the same IP of the server, and the problem persist.
authelia.log
root@ip-10-255-140-166:/etc/authelia# tail authelia.log 
time="2023-11-15T17:53:03Z" level=info msg="Storage schema is being checked for updates"
time="2023-11-15T17:53:03Z" level=info msg="Storage schema is already up to date"
time="2023-11-15T17:53:03Z" level=info msg="Initializing server for non-TLS connections on '[::]:9091' path '/'"
time="2023-11-15T17:53:25Z" level=debug msg="Mark 1FA authentication attempt made by user 'batman'" method=POST path=/api/firstfactor remote_ip=192.168.1.1
time="2023-11-15T17:53:25Z" level=debug msg="Successful 1FA authentication attempt made by user 'batman'" method=POST path=/api/firstfactor remote_ip=192.168.1.1
time="2023-11-15T17:53:25Z" level=debug msg="Check authorization of subject username=batman groups=admins,dev ip=192.168.1.1 and object https://abacaxi.prevnet/ (method )."
time="2023-11-15T17:53:25Z" level=debug msg="Required level for the URL https://abacaxi.prevnet/ is 1" method=POST path=/api/firstfactor remote_ip=192.168.1.1
time="2023-11-15T17:53:25Z" level=debug msg="Redirection URL https://abacaxi.prevnet/ is safe" method=POST path=/api/firstfactor remote_ip=192.168.1.1
time="2023-11-15T17:53:26Z" level=debug msg="Check authorization of subject username= groups= ip=192.168.1.1 and object https://abacaxi.prevnet/ (method )."
time="2023-11-15T17:53:26Z" level=info msg="Access to https://abacaxi.prevnet/ (method unknown) is not authorized to user <anonymous>, responding with status code 401" method=GET path=/api/verify remote_ip=192.168.1.1

192.168.1.1 - IP Authelia Server

**Reply by james-d-elliott** [reply]
It's not overly practical for me to give you too much advice about this as there are so many reasons why this could be wrong which are not related to authelia directly and are rather infrastructure of proxy related.

Things that could be wrong:

- The NAT use of NAT hairpin may be showing the LAN IP of the router.
- The header may be absent from your configuration.
- The header may be changed by the existence of a second proxy such as:
  - Another nginx Server
  - Cloudflare proxy
  
  
 The requests made by `auth_request` to the `/api/verify` endpoint MUST have the correct X-Forwarded-For header. 
 
It's likely from your supplied configurations it's two nginx servers in which case on the authelia nginx you'd have to ensure specifically all requests to /api/verify do not have the X-Forwarded-For changed by either ensuring the proxy server handling the request is trusted by the authelia nginx, or by creating a /api/verify location customized to your environment.

**Comment by josue-soares** [comment]
My files (test environment)
[etc-authelia.tar.gz](https://github.com/authelia/authelia/files/13378001/etc-authelia.tar.gz)
[etc-nginx-authelia-server.tar.gz](https://github.com/authelia/authelia/files/13378002/etc-nginx-authelia-server.tar.gz)
[etc-nginx-target-app.tar.gz](https://github.com/authelia/authelia/files/13378003/etc-nginx-target-app.tar.gz)

**Comment by josue-soares** [comment]
Do you have a models of files to config nginx proxy?

**Reply by james-d-elliott** [reply]
https://www.authelia.com/integration/proxies/nginx/
